### PR TITLE
fix: ANR when recording an audio (WPB-9528)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9528" title="WPB-9528" target="_blank">BugWPB-9528</a>  ANR when recording an audio 
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

ANR happens when user try to record an audio

### Causes (Optional)

Some oprations related to recording audio is are done on Main thread

### Solutions

Change the audio recording to be done on worker thread instead 

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [5126](https://10.10.124.134/job/AR-build-pipeline/5126/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.1(9846312)(9846312)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.1(9846312)(9846312)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt [hunk 1]
```diff
@@ -169,31 +171,33 @@ class RecordAudioViewModel @Inject constructor(
     }
 
     fun stopRecording() {
-        if (state.buttonState == RecordAudioButtonState.RECORDING) {
-            audioMediaRecorder.stop()
-        }
-        audioMediaRecorder.release()
+        viewModelScope.launch(dispatchers.default()) {
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt:174]: can we keep the `viewModelScope.launch` as it is and switch dispatchers for generateAudioFileWithEffects here

https://github.com/wireapp/wire-android/blob/7b1afb541948d8761b22e871bb03cc9354e2ec4d/app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/GenerateAudioFileWithEffectsUseCase.kt#L39

Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt:174]: It's not only for `generateAudioFileWithEffects`, we need also to run `getPlayableAudioFile()`, and `getAudioLengthInMs()`and ` stop()` in worker thread

Comment (MohamadJaara) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioViewModel.kt:174]: Then we can change of them individual, updating the state from a background thread can cause some issues.

Happy to talk about this in a 1:1 or in the next collective

**Review by MohamadJaara — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Pull request merged**